Week-3/Exercise3.1/js/index.js

```diff
@@ -0,0 +1,29 @@
+function add(a,b){
+    return a + (b || 0);
+}
+
+
+const memoizeAdd = memoize(add);
+
+function memoize(reducer) {
+
+  function add (...inputs) {
+
+    if (memoize.cache[inputs]) {
+      return memoize.cache[inputs];
+    }
+    
```

Comment:
Good implementation but your memoize function is only checking for inputs. It should also check and cache function name, so that it works for add, substract etc